apps/dashboard/app/(app)/apis/[apiId]/keys/[keyAuthId]/[keyId]/reroll-key.tsx

```diff
@@ -0,0 +1,195 @@
+"use client";
+
+import { Loading } from "@/components/dashboard/loading";
+import { Button } from "@/components/ui/button";
+import {
+  Dialog,
+  DialogContent,
+  DialogDescription,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+  DialogTrigger,
+} from "@/components/ui/dialog";
+import {
+  Form,
+  FormDescription,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+} from "@/components/ui/form";
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from "@/components/ui/select";
+import { trpc } from "@/lib/trpc/client";
+import { parseTrpcError } from "@/lib/utils";
+import { zodResolver } from "@hookform/resolvers/zod";
+import type { Key } from "@unkey/db";
+import { useRouter } from "next/navigation";
+import { useState } from "react";
+import { useForm } from "react-hook-form";
+import { toast } from "sonner";
+import { z } from "zod";
+
+type Props = {
+  trigger: React.ReactNode;
+  currentKey: Key;
+  apiId: string;
+};
+
+const EXPIRATION_OPTIONS = [
+  { key: "now", value: "Now" },
+  { key: "5m", value: "5 minutes" },
+  { key: "30m", value: "30 minutes" },
+  { key: "1h", value: "1 hour" },
+  { key: "6h", value: "6 hours" },
+  { key: "24h", value: "24 hours" },
+  { key: "7d", value: "7 days" },
+];
+
+const formSchema = z.object({
+  expiresIn: z.coerce.string(),
+});
+
+export const RerollKey: React.FC<Props> = ({ trigger, currentKey, apiId }: Props) => {
+  const router = useRouter();
+  const [open, setOpen] = useState(false);
+  const { keyAuthId } = currentKey;
+  const [newKeyId, setNewKeyId] = useState<string | null>();
+
+  const form = useForm<z.infer<typeof formSchema>>({
+    resolver: zodResolver(formSchema),
+    reValidateMode: "onBlur",
+    defaultValues: {
+      expiresIn: "1h",
+    },
+  });
+
+  const createKey = trpc.key.create.useMutation({
+    onSuccess({ keyId }) {
+      toast.success("Rerolling in progress.", {
+        description: "Step 1/2",
+      });
+      setNewKeyId(keyId);
+    },
+    onError(err) {
+      console.error(err);
+      const message = parseTrpcError(err);
+      toast.error(message);
+    },
+  });
+
+  const updateDeletedAt = trpc.key.update.deletedAt.useMutation({
+    onSuccess() {
+      toast.success("Rerolling completed.", {
+        description: "Step 2/2",
```

Comment:
The "Step X/X" doesn't add much value I think, wdyt?

Reply:
yes, agreed. I'll remove it